io/io/src/TKey.cxx

```diff
@@ -477,7 +477,10 @@ void TKey::Create(Int_t nbytes, TFile* externFile)
             nsize,GetName(),GetTitle());
       return;
    }
-   fDatime.Set();
+   if (f->TestBit(TFile::kReproducible))
+      fDatime = (UInt_t) 1; // cannot use 0, when reading from file 0 time will be reassigned
+   else
+      fDatime.Set();
```

Comment:
@pcanal 
This is the only place where data member has to be set directly.
In several places temporary TKey object is created and TFile is not assigned to it.
Therefore when filling buffer TKey has no idea that file has such "predictable" feature
I guess it should be ok for now

Reply:
An alternative might be
```
if (f->TestBit(TFile::kReproducible))
  SetBit(TKey::kReproducible)
```

Reply:
But this extra bit in TKey is useless.
It only required for temporary TKey objects, used to write some TBasket or TFile data into the file. Probably, instead of `TKey::kReproducible` bit (which will be stored in file) one can just use extra non-persistent boolean member.

Reply:
As far as I remember the bits are not stored for a TKey, am I remembering wrong?

Reply:
Yes, you are right. I add correspondent commits